File: nos2jdbc-tutorial-spring/src/main/java/nos2jdbc/tutorial/spring/entity/nonauto/MemberItem.java
package nos2jdbc.tutorial.spring.entity.nonauto;

import java.math.BigDecimal;
import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Transient;
import nos2jdbc.annotation.NoFk;
import nos2jdbc.annotation.NonAuto;

@Entity @NonAuto
public class MemberItem {
    @Id
    public Long lunchFeeId;
    public LocalDate payOn;
    public BigDecimal amount;
    public int count;
    
    @Transient @ManyToOne @NoFk
    public MemberYm ym;
 }

File: nos2jdbc-tutorial-kotlin-spring/src/main/kotlin/nos2jdbc/tutorial/kotlinspring/entity/nonauto/MemberItem.kt
package nos2jdbc.tutorial.kotlinspring.entity.nonauto

import java.math.BigDecimal
import java.time.LocalDate

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.Transient

import nos2jdbc.annotation.NoFk
import nos2jdbc.annotation.NonAuto

@Entity @NonAuto
class MemberItem {
    @Id
    var lunchFeeId: Long? = null
    var payOn: LocalDate? = null
    var amount: BigDecimal = BigDecimal.ZERO
    var count: Int = 0
    
    @Transient @ManyToOne @NoFk
    var ym: MemberYm? = null
 }
